### PR TITLE
Moved duplicate code to BuildingBlocks shared lib

### PR DESCRIPTION
**Pull request by Dannyps**
# Readiness checklist

- [ ] I added/updated unit tests.
- [ ] I added/updated integration tests.
- [x] I ensured that the PR title is good enough for the changelog.
- [x] I labeled the PR.

### CODE REVIEW

**Comment by Dannyps** [comment]
@tnotheis I've opening with PR with just a small change so that you confirm if this is the desired approach. Thanks

**Review by tnotheis — COMMENTED** [review]
Almost. But don't move the stuff to BuildingBlocks.Application. The Application projects should not know anything about the "outside world", like the database, the blob storage, the event bus - and not about the HTTP API either.

File: BuildingBlocks/src/BuildingBlocks.Application/BuildingBlocks.Application.csproj [hunk 1]
```diff
@@ -11,7 +11,9 @@
 		<PackageReference Include="ByteSize" Version="2.1.1" />
 		<PackageReference Include="FluentValidation" Version="11.5.2" />
 		<PackageReference Include="MediatR" Version="12.0.1" />
+		<PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
 		<PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.5" />
+		<PackageReference Include="Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions" Version="6.0.2" />
```
Comment (tnotheis) [inline BuildingBlocks/src/BuildingBlocks.Application/BuildingBlocks.Application.csproj:16]: Remember to remove the added references here

File: BuildingBlocks/src/BuildingBlocks.API/Mvc/JsonConverters/UtcDateTimeConverter.cs [hunk 2]
```diff
@@ -8,11 +8,16 @@ public class UtcDateTimeConverter : JsonConverter<DateTime>
...
-    public UtcDateTimeConverter(string format = DEFAULT_FORMAT)
+    public UtcDateTimeConverter(string format)
     {
         _format = format;
     }
 
+    public UtcDateTimeConverter()
+    {
+        _format = DEFAULT_FORMAT;
+    }
```
Comment (tnotheis) [inline BuildingBlocks/src/BuildingBlocks.API/Mvc/JsonConverters/UtcDateTimeConverter.cs:19]:
```suggestion
    public UtcDateTimeConverter() : base(DEFAULT_FORMAT) { }
```

File: BuildingBlocks/src/BuildingBlocks.API/Mvc/JsonConverters/UtcDateTimeConverter.cs [hunk 3]
```diff
@@ -38,10 +43,14 @@ public class NullableUtcDateTimeConverter : JsonConverter<DateTime?>
...
 
-    public NullableUtcDateTimeConverter(string format = UtcDateTimeConverter.DEFAULT_FORMAT)
+    public NullableUtcDateTimeConverter(string format)
     {
         _format = format;
     }
+    public NullableUtcDateTimeConverter()
+    {
+        _format = UtcDateTimeConverter.DEFAULT_FORMAT;
+    }
```
Comment (tnotheis) [inline BuildingBlocks/src/BuildingBlocks.API/Mvc/JsonConverters/UtcDateTimeConverter.cs:53]:
```suggestion
    public NullableUtcDateTimeConverter() : base(UtcDateTimeConverter.DEFAULT_FORMAT) { }
```

**Review by tnotheis — CHANGES_REQUESTED** [review]

**Review by tnotheis — APPROVED** [review]

**Pull request merged**